### DISCUSSION TITLE
modulenotfounderror: no module named 'discord'

### DISCUSSION THREAD

**Discussion by BreakableTuber** [Rapptz/discord.py#6376] [General] [answered]
Traceback (most recent call last):
  File "C:\Users\gefor\Downloads\Shindo_Life_Fans\bot.py", line 1, in <module>
    import discord
ModuleNotFoundError: No module named 'discord'

the import is
-----------------------------------------------------------------------------------------

import discord
from discord.ext import commands

**Comment by tk421bsod** [comment]
You need to install discord.py before you can use it. You can install it using `python -m pip install -U discord.py`. Run that command where you're going to be running your bot from. This is because some IDEs (like PyCharm) create [virtual environments](https://docs.python.org/3/tutorial/venv.html) for each project you create. This can lead to ModuleNotFound errors because packages installed inside of those virtual environments aren't accessible from outside those environments. (or vice versa)

**Reply by BreakableTuber** [reply]
didn't work, i use visual studio, and when i ran the command in the folder my bot.py is in it said Requirement already satisfied, skipping upgrade

**Reply by BreakableTuber** [reply]
Requirement already satisfied: discord.py in c:\users\gefor\appdata\local\programs\python\python38\lib\site-packages (1.6.0)
Requirement already satisfied: aiohttp<3.8.0,>=3.6.0 in c:\users\gefor\appdata\local\programs\python\python38\lib\site-packages (from discord.py) (3.7.3)
Requirement already satisfied: chardet<4.0,>=2.0 in c:\users\gefor\appdata\local\programs\python\python38\lib\site-packages (from aiohttp<3.8.0,>=3.6.0->discord.py) (3.0.4)
Requirement already satisfied: yarl<2.0,>=1.0 in c:\users\gefor\appdata\local\programs\python\python38\lib\site-packages (from aiohttp<3.8.0,>=3.6.0->discord.py) (1.6.3)
Requirement already satisfied: multidict<7.0,>=4.5 in c:\users\gefor\appdata\local\programs\python\python38\lib\site-packages (from aiohttp<3.8.0,>=3.6.0->discord.py) (5.1.0)
Requirement already satisfied: attrs>=17.3.0 in c:\users\gefor\appdata\local\programs\python\python38\lib\site-packages (from aiohttp<3.8.0,>=3.6.0->discord.py) (20.3.0)
Requirement already satisfied: typing-extensions>=3.6.5 in c:\users\gefor\appdata\local\programs\python\python38\lib\site-packages (from aiohttp<3.8.0,>=3.6.0->discord.py) (3.7.4.3)
Requirement already satisfied: async-timeout<4.0,>=3.0 in c:\users\gefor\appdata\local\programs\python\python38\lib\site-packages (from aiohttp<3.8.0,>=3.6.0->discord.py) (3.0.1)
Requirement already satisfied: idna>=2.0 in c:\users\gefor\appdata\local\programs\python\python38\lib\site-packages (from yarl<2.0,>=1.0->aiohttp<3.8.0,>=3.6.0->discord.py) (3.1)

**Comment by BreakableTuber** [comment, marked as answer]
fixed it, had like 20 million fake discord n 20 files of python, and deleted it all, reinstalled k ty tho
